luigi/mock.py

```diff
@@ -158,7 +158,7 @@ def write(self, data):
                 super(Buffer, self).write(data)
 
             def close(self):
-                if mode == 'w':
+                if mode[0] == 'w':
```

Comment:
is it possible for `mode` to be `None`? If so, this will throw an exception

Reply:
From python2 and python3 when we open a file we use it like `f = open("file.extension", mode)` where mode should be a string, if not it should throw an exception, should I add a test for this also?

Reply:
Python allows you to `with open("file.extension")` where the mode is defaulted to `r`. However, i'm just not clear on the behavior here given we're overriding `open`. It wouldn't hurt to have  a test for sanity :)

Reply:
Yes, I am going to write more tests about it and notify then. Thank you for the comments :)

Reply:
@dlstadther I refactored the byte test and added and default open action as you mentioned in the discussion and added an error check test if type is None 👍